### PR TITLE
Add/update IE/EdgeHTML data for Document API

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR updates the Internet Explorer and pre-Blink Edge data for the Document API based upon manual testing to achieve more real data for the Document API.

Tests used: https://mdn-bcd-collector.appspot.com/tests/api/Document

### CODE REVIEW

File: api/Document.json [hunk 1]
```diff
@@ -1528,7 +1528,7 @@
               "version_added": true
             },
             "ie": {
-              "version_added": null
+              "version_added": "9"
```
Comment (foolip) [inline api/Document.json:1531]: This surprised me, but indeed it seems like `document.createAttributeNS` isn't in IE8. The documents we're testing in are HTMLDocument instances, and it's possible `createAttributeNS` was on XMLDocument, but IE8 actually doesn't seem to have XMLDocuemnt.

https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/ff975561(v=vs.85) seems to support this change.

File: api/Document.json [hunk 2]
```diff
@@ -2988,7 +2988,7 @@
               "version_added": true
             },
             "ie": {
-              "version_added": true
+              "version_added": false
```
Comment (foolip) [inline api/Document.json:2991]: This is consistent with `document.documentURI` being added in Edge 17.

File: api/Document.json [hunk 3]
```diff
@@ -7660,7 +7660,7 @@
               "version_added": "56"
             },
             "ie": {
-              "version_added": "10"
+              "version_added": false
```
Comment (foolip) [inline api/Document.json:7663]: This is `document.onvisibilitychange`, where the data claimed that it was in IE 10.

https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event#Browser_compatibility (just edited) says the event itself was in IE 10. I'm not sure which is correct, but can you back out this change and update both separately? The event can't be tested by mdn-bcd-collector, and it's just as well to separate out such changes.

File: api/Document.json [hunk 4]
```diff
@@ -7769,7 +7769,7 @@
               "version_added": false
             },
             "ie": {
-              "version_added": true
+              "version_added": false
```
Comment (foolip) [inline api/Document.json:7772]: I was a bit skeptical of this, so tested it separate in https://software.hixie.ch/utilities/js/live-dom-viewer/?saved=8540 using IE11. But `document.origin` seems to not have been in IE, so this change is correct.

File: api/Document.json [hunk 5]
```diff
@@ -10150,7 +10150,7 @@
               }
             ],
             "ie": {
-              "version_added": true
+              "version_added": false
```
Comment (foolip) [inline api/Document.json:10153]: Yeah, the true there wasn't plausible, given that Edge didn't have `document.timeline`.

File: api/Document.json (hunk 3 above)
Comment (vinyldarkscratch) [reply api/Document.json:7663]: It looks like IE 10 didn't support `onvisibilitychange`, however it supported the event through `addEventListener`.  Reverting these changes and cherry-picking in a separate PR.

Comment (vinyldarkscratch) [reply api/Document.json]: Cherry-picked as #6833.

**Review by foolip — APPROVED** [review]

**Pull request merged**